engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/syntax/HostValueToEnsoNode.java

```diff
@@ -65,7 +72,43 @@ Atom doNull(
   }
 
   @Fallback
-  Object doOther(Object o) {
+  Object doOther(Object o, @CachedLibrary(limit = "3") InteropLibrary dates) {
+    if (o instanceof Atom) {
+      return o;
+    }
+    if (dates.isDate(o)) {
+      CompilerDirectives.transferToInterpreter();
+      final RootNode rn = getRootNode();
+      if (rn != null) {
+        final SourceSection ss = rn.getSourceSection();
+        if (ss != null) {
+          final Source s = ss.getSource();
+          if (s.getName().equals("Date.enso")) {
```

Comment:
A chicken and egg problem, @jdunkerley & @kustosz!

The request is to convert all incoming `java.time.LocalDate` to Enso's `Standard.Base.Data.Time.Date.Date` - but the Enso `Standard.Base.Data.Time.Date` is actually making calls into `java.time.LocalDate` methods and does need the unconverted reference!

Vicious circle showing a significant flaw...